### PR TITLE
Added AdminStudentCheckInView

### PR DESCRIPTION
**Pull request by danmconrad**
Fixes https://github.com/CoderDojoChi/coderdojochi/issues/502

- [x] Adds the `coderdojochi` build folder to `.gitignore`
- [x] Removes `session_check_in` from `old_views.py`
- [x] Creates a base `AdminView` to abstract cache, login, and redirect for subsequent views
- [x] Creates `AdminStudentCheckInView` based on all of the old logic

### CODE REVIEW

**Comment by danmconrad** [comment]
Preview of the page working:
student-check-in

File: src/coderdojochi/views/admin/student_check_in.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+
+class AdminStudentCheckInView(AdminView):
+    template_name = 'session-check-in.html'
+
+    def get_context_data(self, **kwargs):
+        session_id = kwargs['session_id']
+        session = get_object_or_404(Session, id=session_id)
+
+        print(session)
```
Comment (karbassi) [inline src/coderdojochi/views/admin/student_check_in.py]: Remove debug code.

File: src/coderdojochi/views/admin/__init__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import never_cache
+from django.views.generic import TemplateView
+
+
+@method_decorator(never_cache, name='dispatch')
+@method_decorator(login_required, name='dispatch')
+class AdminView(TemplateView):
+    def dispatch(self, request, *args, **kwargs):
```
Comment (karbassi) [inline src/coderdojochi/views/admin/__init__.py]: 👍 Great use of `dispatch`! 🌟

File: .gitignore [hunk 3]
```diff
@@ -30,3 +30,5 @@ Thumbs.db
 *.sql
 
 # End of Miscellaneous
+
+coderdojochi
```
Comment (karbassi) [inline .gitignore]: I don't think we need this in here.

File: src/coderdojochi/views/admin/student_check_in.py (hunk 1 above)
Comment (danmconrad) [reply src/coderdojochi/views/admin/student_check_in.py]: gah!, thanks!

**Pull request merged**